agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/api/resource/ebs_discovery.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+)
+
+const (
+	ebsnvmeIDTimeoutDuration = 5 * time.Second
+	ebsResourceKeyPrefix     = "ebs-volume:"
+	ScanPeriod               = 500 * time.Millisecond
+)
+
+var (
+	ErrInvalidVolumeID = errors.New("EBS volume IDs do not match")
+)
+
+// EBSDiscoveryClient is used to scan and validate if an EBS volume is attached on the host instance.
+type EBSDiscoveryClient struct {
+	ctx context.Context
+}
+
+// ScanTickerController is used to keep track of the scanning interval of EBS volumes for the EBS watcher.
+type ScanTickerController struct {
+	ScanTicker *time.Ticker
+	Running    bool
+	TickerLock sync.Mutex
+	Done       chan bool
+}
```

Comment:
There is too much state in this struct.... can we keep the ticker running and call `checkPendingEBSVolumes` function in the tick handler to decide if a scan should be performed for this tick? If not, then we do nothing in this tick. The runtime cost of having a ticker running is much lower than the readability and maintenance cost of ticker lifecycle management IMO.

Reply:
No longer using the `ScanTickerController`. We'll have the watcher to start up and have the ticker to always be ticking as long as agent is up. Thanks!